.github/workflows/build-with-clang.yml

```
@@ -67,12 +68,12 @@ jobs:
           export CC=$CMPLR_ROOT/bin/icx
           export CXX=$CMPLR_ROOT/bin/icpx
           export CFLAGS="${CFLAGS} -fno-fast-math -O2"
-          python setup.py develop
+          pip install -e . --no-build-isolation --no-deps --verbose
 
       - name: Run mkl_random tests
         shell: bash -l {0}
         run: |
           source /opt/intel/oneapi/setvars.sh
           # Test with NumPy<2 for now
-          pip install numpy"<2"
+          pip install numpy"<2" pytest
```

I believe we can relax this condition and test with the same numpy version that is used in build process.
```suggestion
          pip install pytest
```